lib/inspec/resources/oracle.rb

```diff
@@ -0,0 +1,65 @@
+# copyright: 2015, Vulcano Security GmbH
+
+require "inspec/resources/powershell"
+
+module Inspec::Resources
+  class Oracle < Inspec.resource(1)
+    name "oracle"
+    supports platform: "unix"
+    supports platform: "windows"
+
+    desc "The 'oracle' resource is a helper for the 'oracle_listener_conf'"
+
+    attr_reader :conf_path
+
+    def initialize
+      case inspec.os[:family]
+      when "debian", "redhat", "linux", "suse"
+        determine_conf_dir_and_path_in_linux
+      when "windows"
+        determine_conf_dir_and_path_in_windows
+      end
+    end
+
+    def to_s
+      "OracleDB"
+    end
+
+    private
+
+    def determine_conf_dir_and_path_in_linux
+      conf_files = inspec.command("sudo find / -type f -wholename '*network/admin/listener.ora'").stdout.lines
```

Comment:
There are a couple of issues with this.

First, we can't issue a sudo statement. We don't know what the sudo password is, and even if sudo is permitted for the command given. In fact, if we are running with the --sudo option to inspec exec, this will end up being wrapped in double sudo.

The second issue is that we really shouldn't run a find command on `/`. What are the likely locations? If there are no likely locations, can we use an option or environment variable? What is $ORACLE_HOME - why isn't that being used here?

Reply:
Yes $ORACLE_HOME is an environment variable which I can try using for finding the file. Will make this change.